worldedit-bukkit/src/main/java/com/sk89q/worldedit/bukkit/adapter/impl/FAWE_Spigot_v1_16_R1.java

```diff
@@ -458,4 +405,434 @@ public int getInternalBiomeId(BiomeType biome) {
+        private Convertable.ConversionSession session;
+        private DefinedStructureManager structureManager;
+        private LightEngineThreaded lightEngine;
+        private ChunkGenerator generator;
+
+        private Path tempDir;
+
+        private boolean generateFlatBedrock = false;
+
+        public RegeneratorImpl(org.bukkit.World originalBukkitWorld, Region region, Extent target, RegenOptions options) {
+            super(originalBukkitWorld, region, target, options);
+        }
+
+        @Override
+        protected boolean prepare() {
+            this.originalNMSWorld = ((CraftWorld) originalBukkitWorld).getHandle();
+            originalChunkProvider = originalNMSWorld.getChunkProvider();
+            if (!(originalChunkProvider instanceof ChunkProviderServer)) {
+                return false;
+            }
+
+            //flat bedrock? (only on paper)
+            try {
+                generateFlatBedrock = flatBedrockField.getBoolean(worldPaperConfigField.get(originalNMSWorld));
+            } catch (Exception ignored) {
+            }
+            
+            seed = options.getSeed().orElse(originalNMSWorld.getSeed());
+            chunkStati.forEach((s, c) -> super.chunkStati.put(new ChunkStatusWrap(s), c));
+
+            return true;
+        }
+
+        @Override
+        protected boolean initNewWorld() throws Exception {
+            //world folder
+            tempDir = java.nio.file.Files.createTempDirectory("WorldEditWorldGen");
+
+            //prepare for world init (see upstream implementation for reference)
+            org.bukkit.World.Environment env = originalBukkitWorld.getEnvironment();
+            org.bukkit.generator.ChunkGenerator gen = originalBukkitWorld.getGenerator();
+            Convertable convertable = Convertable.a(tempDir);
+            ResourceKey<WorldDimension> worldDimKey = getWorldDimKey(env);
+            session = convertable.c("worldeditregentempworld", worldDimKey);
+            WorldDataServer originalWorldData = originalNMSWorld.worldDataServer;
+
+            MinecraftServer server = originalNMSWorld.getServer().getServer();
+            WorldDataServer levelProperties = (WorldDataServer) server.getSaveData();
+            RegistryReadOps<NBTBase> nbtRegOps = RegistryReadOps.a(DynamicOpsNBT.a, server.dataPackResources.h(), IRegistryCustom.b());
+            GeneratorSettings newOpts = GeneratorSettings.a.encodeStart(nbtRegOps, levelProperties.getGeneratorSettings()).flatMap(tag -> GeneratorSettings.a.parse(this.recursivelySetSeed(new Dynamic<>(nbtRegOps, tag), seed, new HashSet<>()))).result().orElseThrow(() -> new IllegalStateException("Unable to map GeneratorOptions"));
+            WorldSettings newWorldSettings = new WorldSettings("worldeditregentempworld", originalWorldData.b.getGameType(), originalWorldData.b.hardcore, originalWorldData.b.getDifficulty(), originalWorldData.b.e(), originalWorldData.b.getGameRules(), originalWorldData.b.g());
```

Comment:
I'm not sure howto solve that properly, but the (bukkit) world (name) that is delegated to custom chunk generators must match the original name, otherwise plugins that load settings based on the world name will fail (aplies for all versions obviously).

Reply:
having 2 worlds with the same name is not possible, also upstream seems to do the same so i guess thats fine?

Reply:
It's not fine as it's breaking compatibility with generators/populators that expect a specific world name like PlotSquared does. I don't know if WE breaks with PlotSquared, but as the bukkit ChunkGenerator and BlockPopulator give direct access to the world object, it  should also provide the "real" world name.

I was thinking about using an anonymous CraftServer object that overrides the addWorld method and use this to override the getServer method in nms World but this is super hacky. Another attempt would be to only fake the return value of the bukkit world by wrapping it and analyzing the stack trace but that's even worse.

Maybe someone else has an idea that is a bit less insane...

Reply:
I'll ping @Sauilitired and @me4502 for this.

Reply:
I'm unsure of specifics, but from memory Bukkit completely explodes if you try to get around the name requirement issue.

Reply:
Just to make sure, I tested the //regen from WorldEdit in a PlotSquared World. Normal WorldEdit is able to regenerate that without any issues: 
![2020-10-08_10 56 05](https://user-images.githubusercontent.com/21148213/95438115-25be5480-0956-11eb-9354-0e0f66f39616.png)
while currently our implementation is unable to regenerate anything in this world, just throwing a NPE.

Reply:
Presumably FAWE is not correctly accessing the generators for the world it's attempting to regenerate

Reply:
The generators are accessed correctly but PlotSquared fails to load the internal PlotArea, as it uses the name `worldeditregentempworld`. I'll try to figure out why it works with WE.

Reply:
Works now, tested on 1.15.2 and 1.16.3